pylint/checkers/similar.py

```diff
@@ -26,32 +26,315 @@
+    Return two dicts. The first associates the hash of successive stripped lines of a lineset
+    to the indices of the starting lines.
+    The second dict, associates the index of the starting line in the lineset's stripped lines to the
+    couple [start, end] lines number in the corresponding file.
+
+    :param lineset: lineset object (i.e the lines in a file)
+    :param min_common_lines: number of successive lines that are used to compute the hash
+    :return: a dict linking hashes to corresponding start index and a dict that links this
+             index to the start and end lines in the file
+    """
+    hash2index = defaultdict(list)
+    index2lines = dict()
+    # Comments, docstring and other specific patterns maybe excluded -> call to stripped_lines
+    # to get only what is desired
+    lines = list(x.text for x in lineset.stripped_lines)
+    # Need different iterators on same lines but each one is shifted 1 from the precedent
+    shifted_lines = [iter(lines[i:]) for i in range(min_common_lines)]
+
+    for index_i, *succ_lines in enumerate(zip(*shifted_lines)):
+        start_linenumber = lineset.stripped_lines[index_i][1]
+        try:
+            end_linenumber = lineset.stripped_lines[index_i + min_common_lines][1]
+        except IndexError:
+            end_linenumber = lineset.stripped_lines[-1][1] + 1
+
+        index = Index(index_i)
+        index2lines[index] = SuccessiveLinesLimits(
+            LineNumber(start_linenumber), LineNumber(end_linenumber)
+        )
+
+        successive_lines = tuple(*succ_lines)
+        l_c = LinesChunk(lineset.name, index, *successive_lines)
+        hash2index[l_c].append(index)
+
+    return hash2index, index2lines
+
+
+def remove_successives(all_couples: CplIndexToCplLines_T) -> None:
+    """
+    Removes all successive entries in the dictionary in argument
+
+    :param all_couples: collection that has to be cleaned up from successives entries.
+                        The keys are couples of indices that mark the beginning of common entries
+                        in both linesets. The values have two parts. The first one is the couple
+                        of starting and ending line numbers of common successives lines in the first file.
+                        The second part is the same for the second file.
+
+    For example consider the following dict:
+
+    >>> all_couples
+    {(11, 34): ([5, 9], [27, 31]),
+     (23, 79): ([15, 19], [45, 49]),
+     (12, 35): ([6, 10], [28, 32])}
+
+    There are two successives keys (11, 34) and (12, 35).
+    It means there are two consecutive similar chunks of lines in both files.
+    Thus remove last entry and update the last line numbers in the first entry
+
+    >>> remove_successives(all_couples)
+    >>> all_couples
+    {(11, 34): ([5, 10], [27, 32]),
+     (23, 79): ([15, 19], [45, 49])}
+    """
+    all_couples_keys: List[LineSetStartCouple] = list(all_couples.keys())
+    for couple in all_couples_keys:
```

Comment:
You don't need to create a temporary list to iterate over `all_couples.keys()`

Reply:
I think if required the typing can be on the for loop like this:
```python
couple: LineSetStartCouple
for couple in all_couples_keys:
    ...
```
https://stackoverflow.com/a/41641489/2519059

Reply:
In fact it is not possible here because the `all_couples` dictionary is modified inside the `for` loop. 
Doing this will raise a `RuntimeError: dictionary changed size during iteration`.

Reply:
@hippo91 👍🏻 Didn't saw that the first time. In that case it should be possible to do
```py
for couple in list(all_couples.keys()):
    ...
```
as `all_couples_keys` isn't used anywhere else.